api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java

```diff
@@ -730,4 +731,65 @@ public boolean isOrderTypeInUse(OrderType orderType) {
 		criteria.add(Restrictions.eq("encounter", encounter));
 		return criteria.list();
 	}
+
+	/**
+	 * @see org.openmrs.api.db.OrderDAO#getOrderGroupAttributeByUuid(String)
+	 */
+	@Override
+	public OrderGroupAttribute getOrderGroupAttributeByUuid(String uuid)  throws DAOException{
+	return (OrderGroupAttribute) sessionFactory.getCurrentSession().createQuery("from OrderGroupAttribute d where d.uuid = :uuid")
+			.setString("uuid", uuid).uniqueResult();
+	}
+
+	/**
+	 * @see org.openmrs.api.db.OrderDAO#getAllOrderGroupAttributeTypes()
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public List<OrderGroupAttributeType> getAllOrderGroupAttributeTypes() throws DAOException{
+		return sessionFactory.getCurrentSession().createCriteria(OrderGroupAttributeType.class).list();
+	}
+
+	/**
+	 * @see org.openmrs.api.db.OrderDAO#getOrderGroupAttributeType(java.lang.Integer)
```

Comment:
Should the method name end with ById?

Reply:
Resolved .

@dkayiwa  do you think something similar here also needs to be changed ?
https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/db/OrderDAO.java

Reply:
@dkayiwa  did alter the method name including the param name to align with method it's self.

Reply:
I looked at the convention and fixed it to  **getOrderGroupAttributeType**(Integer orderGroupAttributeTypeId)